text/0000-update_asset_packaging.md

```diff
@@ -0,0 +1,61 @@
+The node modules will be integrated into foreman using Webpack. Webpack has a powerful system of loaders that allow handling various tasks. Webpack also comes with a small webserver that allows live-reload of all assets managed by it, hopefully leading to increased developer happiness. In addition, webpack has various other strengths that make it seem like the current leading option - ability to handle css assets and all js module flavors, smart division of code into bundles and more.
+As we will currently continue using the Rails assets pipeline, any plugin will be able to use whatever method they choose to handle any assets as long as the result is added to the pipeline (as is done currently).
+After a certain period of attempting this solution, we will review the choice and decide if we wish to continue in this direction or change to a different one.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Some reasons to stay with the current solution:
+* No extra work needed.
+* Single package management system - bundler. All dependencies are in one central location, build process is simple.
+* Rails asset pipeline doesn’t require any modifications to handle the current solution.
+
+# Alternatives
+[alternatives]: #alternatives
+
+Several other alternatives were considered before choosing this option. The alternatives considered:
+
+For package management - NPM, Bower, remain with current solution.
+
+For handling the assets (getting them working with foreman) - Grunt, Gulp, Webpack, Browserify, Rake.
+
+History of the discussion and pros and cons of each option can be read in the previous commits of this file.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+* How to package rpms without internet connection, as Koji does not allow internet connection?
+  * The compiled assets could be committed into source control. This has a downside of committing a large amount of generated code into git, and may lead to issues with forgetting to recompile before commit.
+  * The `node_modules` directory could be added to source control. This again leads to a large amount of files that aren't required polluting our repository and issues with keeping the directory up to date.
+  * Each node module could be built as a separate package. This is unfeasible, since there are hundreds of modules required for the generation of the minified files. In addition, we would not want to package every single one ourselves, given that one of the goals of this change is to reduce packaging workload.
+  * The `node_modules` directory could be added to the source tarball before being sent to the builder, with the compiling and minification being run on the builder.
+  * The js assets could be compiled and minified before the build and have only the resulting files be added into the tarball passed into the builder.
+  * Some concern has been raised that any option other then packaging every single library may go against the Fedora Packaging Guidelines.
```

Comment:
I'm not sure how committed we should be to the Fedora guidelines, since:

- There is no plan to distribute Foreman in Fedora in the foreseeable future.
- We already include various js libraries in the `vendor/`directory, not sure why moving some to `node_modules` would change our compliance.